x-pack/plugins/enterprise_search/public/applications/app_search/components/engine_creation/engine_creation.tsx

```diff
@@ -0,0 +1,123 @@
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React from 'react';
+
+import {
+  EuiForm,
+  EuiFlexGroup,
+  EuiFormRow,
+  EuiFlexItem,
+  EuiFieldText,
+  EuiSelect,
+  EuiPageHeader,
+  EuiPageHeaderSection,
+  EuiSpacer,
+  EuiText,
+  EuiTitle,
+  EuiButton,
+  EuiPanel,
+} from '@elastic/eui';
+import { useActions, useValues } from 'kea';
+
+import { SetAppSearchChrome as SetPageChrome } from '../../../shared/kibana_chrome';
+import { FlashMessages } from '../../../shared/flash_messages';
+import {
+  ALLOWED_CHARS_NOTE,
+  ENGINE_CREATION_FORM_ENGINE_LANGUAGE_LABEL,
+  ENGINE_CREATION_FORM_ENGINE_NAME_LABEL,
+  ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER,
+  ENGINE_CREATION_FORM_SUBMIT_BUTTON_LABEL,
+  ENGINE_CREATION_FORM_TITLE,
+  ENGINE_CREATION_TITLE,
+  SANITIZED_NAME_NOTE,
+  SUPPORTED_LANGUAGES,
+} from './constants';
+import { EngineCreationLogic } from './engine_creation_logic';
+
+export const EngineCreation: React.FC = () => {
+  const { name, rawName, language } = useValues(EngineCreationLogic);
+  const { setLanguage, setRawName, submitEngine } = useActions(EngineCreationLogic);
+
+  return (
+    <div data-test-subj="EngineCreation">
+      <SetPageChrome trail={[ENGINE_CREATION_TITLE]} />
+      <EuiPageHeader>
+        <EuiPageHeaderSection>
+          <EuiTitle size="l">
+            <h1>{ENGINE_CREATION_TITLE}</h1>
+          </EuiTitle>
+        </EuiPageHeaderSection>
+      </EuiPageHeader>
+      <FlashMessages />
+      <EuiPanel>
+        <EuiForm>
+          <form
+            data-test-subj="EngineCreationForm"
+            onSubmit={(e) => {
+              e.preventDefault();
+              submitEngine();
+            }}
+          >
+            <EuiTitle>
+              <EuiText>{ENGINE_CREATION_FORM_TITLE}</EuiText>
+            </EuiTitle>
+            <EuiSpacer />
+            <EuiFlexGroup>
+              <EuiFlexItem>
+                <EuiFormRow
+                  label={ENGINE_CREATION_FORM_ENGINE_NAME_LABEL}
+                  helpText={
+                    name.length > 0 && rawName !== name ? (
+                      <>
+                        {SANITIZED_NAME_NOTE} <strong>{name}</strong>
+                      </>
+                    ) : (
+                      ALLOWED_CHARS_NOTE
+                    )
+                  }
+                  fullWidth={true}
+                >
+                  <EuiFieldText
+                    name="engine-name"
+                    value={rawName}
+                    onChange={(event) => setRawName(event.currentTarget.value)}
+                    autoComplete="off"
+                    fullWidth={true}
+                    data-test-subj="EngineCreationNameInput"
+                    placeholder={ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER}
+                    autoFocus={true}
```

Comment:
[optional + would love to get more opinions from folks on this] Do we have a preference to writing boolean props like just `autoFocus` or being specific about `autoFocus={true}`? i.e.,


```suggestion
                    fullWidth
                    data-test-subj="EngineCreationNameInput"
                    placeholder={ENGINE_CREATION_FORM_ENGINE_NAME_PLACEHOLDER}
                    autoFocus
```

@scottybollinger I can't remember, did we ever have a team discussion on this or do we not super mind either way consistency-wise?

Reply:
I probably sound like a broken record... but [there's an eslint rule for this](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md)!

Reply:
Ooh awesome, I'd love to add it! I'm still not quite sure which we actually prefer as a team though, let's bring adding more eslint rules in front of our whole frontend team here in a bit